pkgs/top-level/all-packages.nix

```diff
@@ -10041,7 +10041,7 @@ with pkgs;
   grocy = callPackage ../servers/grocy { };
 
   inherit (callPackage ../servers/nextcloud {})
-    nextcloud23 nextcloud24 nextcloud25;
+    nextcloud nextcloud23 nextcloud24 nextcloud25;
```

Comment:
This won't evaluate properly because of this safety-check in the module: https://github.com/NixOS/nixpkgs/blob/0a6e6cf7e698a6a08a62d8863e2c66b36d5db0d9/nixos/modules/services/web-apps/nextcloud.nix#L721-L729

Also, `pkgs.nextcloud` seems pretty misleading to me, this really sounds like a package-name to me. Can't we call it `mkNextcloudPackage` or something like that?

Reply:
that name comes out of a brief discussion in #212912, I am not opposed to `mkNextcloudPackage`, but would it not be spelled `mkNextcloudDerivation`? is there a canonical way to do this that can be found elsewhere? I am keen to make this feel like the rest of the flake

Reply:
Fine by me. As long as it's clear that it's a function and that the result has something to do with nextcloud ;-)

Reply:
I have some concern about discoverability of `pkgs.mkNextcloudDerivation`, but I agree with your concerns around `pkgs.nextcloud`

Reply:
Perhaps we could mention it in the nextcloud module docs?